usersimcrs/simulator/agenda_based/agenda_based_simulator.py

```diff
@@ -83,33 +84,96 @@ def generate_response(
         # Response generation (intent and slot-values).
         response_intent = None
         response_slot_values = None
-        response_preference = None
 
         self._interaction_model.update_agenda(agent_intent)
         response_intent = self._interaction_model.current_intent
 
+        # TODO: Refactor the part below to private methods, once the logic
+        # is clear.
+
         # Agent wants to elicit preferences.
         if self._interaction_model.is_agent_intent_elicit(agent_intent):
-            # Read entities (slots) from preference_model.
+            # TODO: Extract the slots from agent response on which preferences
+            # are elicited. For now, we just focus on a single slot.
+            slot = None
+
+            # Agent is soliciting preferences on a particular slot.
+            if slot:
+                # TODO: Extract value from agent response.
+                value = None
+                # Agent is asking about a particular slot-value pair, e.g.,
+                # "Do you like action movies?"
+                if value:
+                    (
+                        response_slot,
+                        response_value,
+                    ) = self._preference_model.get_slot_value_preference(
+                        slot, value
+                    )
+                # Agent is asking about value preferences on a given slot, e.g.,
+                # "What movie genre would you prefer?"
+                else:
+                    (
+                        response_slot,
+                        response_value,
+                    ) = self._preference_model.get_slot_preference(slot)
+
+                if response_slot:
+                    response_intent = self._interaction_model.INTENT_DISCLOSE
+                    response_slot_values = [
+                        Annotation(slot=response_slot, value=response_value)
+                    ]
+                else:
+                    response_intent = self._interaction_model.INTENT_DONT_KNOW
+
+            # Agent is eliciting preferences in general, e.g., "What kind of
+            # movies do you like"
+            else:
+                # TODO: Add method to interaction model that picks both a slot
+                # and a preference.
+                pass
+
+            # TODO: These are old comments that I'm not sure how to interpret...
```

Comment:
I suppose these can be removed as a new logic is proposed above, from my understanding, the idea behind `get_next_user_slots` was to determine what you called `response_slot` and `response_value` above based on the agent's intent and annotations.

Reply:
I'll need to check the implementation for KDD paper; created a separate issue for that and removed these comments.